File: PizzaOrderingSystem.Models/Classes/Pizzas/Pizza.cs
using System.Collections.Generic;

namespace PizzaOrderingSystem.Models.Classes.Pizzas
{

    public class Pizza : BaseModel
    {
        public string Type { get; set; }
        public string Image { get; set; }
        public double BasePrice { get; set; }
        public double Totalprice { get; set; }
        public PizzaSauce Sauce { get; set; }
        public PizzaTopping Topping { get; set; }
        public PizzaBase Base { get; set; }
        public PizzaSize Size { get; set; }
        public PizzaCheese Cheese { get; set; }

        public Pizza()
        {
            Image = string.Empty;
            Sauce = new PizzaSauce();
            Topping = new PizzaTopping();
            Base = new PizzaBase();
            Size = new PizzaSize();

        }
    }
}

File: PizzaOrderingSystem.SPA/src/app/models/Pizza/pizza.ts
import { BaseModel } from "../base-model";
import { PizzaCheese } from "./pizza-cheese";
import { PizzaSize } from "./pizza-size";
import { PizzaSuace } from "./pizza-suace";
import { PizzaTopping } from "./pizza-topping";
import { PizzaBase } from "./pizza-base";

export class Pizza extends BaseModel {
   public addedToCart:boolean;
   public quantity:number;
   public size:PizzaSize;
   public base:PizzaBase;
   public topping:PizzaTopping;
   public cheese:PizzaCheese;
   public sauce:PizzaSuace;

   constructor()
   {
    super()
    this.addedToCart= false;
    this.quantity= 0;
    this.size =  new PizzaSize();
    this.base =  new PizzaBase();
    this.cheese =  new PizzaCheese();
    this.topping =  new PizzaTopping();
    this.size =  new PizzaSize();
    this.sauce =  new PizzaSuace();

   }
}
